src/java_binding/src/lib.rs

```diff
@@ -221,22 +222,46 @@ where
     }
 }
 
-pub enum JavaBindingRow {
+pub enum JavaBindingRowKind {
```

Comment:
Better to rename to `JavaBindingRowInner`. `Kind` is usually plain enum without any extra struct stored inside.